main/src/ca/uwaterloo/flix/util/Validation.scala

```diff
@@ -168,15 +117,10 @@ object Validation {
   /**
     * Represents a success `value`.
     */
-  private case class Success[T, E](t: T) extends Validation[T, E] {
+  case class Success[T, E](t: T) extends Validation[T, E] {
```

Comment:
Made this public again. It is easier to pattern match directly on this instead of `Result` when the constructors are equivalent. Also, this is a step in the direction of refactoring `Validation` later to be what it was before `SoftFailure`.